### PR TITLE
Implement srcset support for amp-img in AMP pages

### PR DESCRIPTION
**Pull request by shtukas**
## What does this change?

We would like our users to get a better experience on Google Discover ( https://www.blog.google/products/search/introducing-google-discover/ ), for this we need to support the `srcset` attribute on `amp-img` ( https://amp.dev/documentation/guides-and-tutorials/develop/style_and_layout/art_direction/ ).

This current PR only upgrade the main media images. Other images can be upgraded the same way

The source of the data for us is the existing `imageSources` attribute of the CAPI answer. The current implementation select a small range of sizes for maximum flexibility. For instance

```
[ { src:
     'https://i.guim.co.uk/img/media/ceaf255ad4460ad68cfe8221257c5363150ccdb1/69_178_3342_2006/master/3342.jpg?width=300&quality=45&auto=format&fit=max&dpr=2&s=9c1fe4f27a56822d5da1d248ec44c5ab',
    width: 600 },
  { src:
     'https://i.guim.co.uk/img/media/ceaf255ad4460ad68cfe8221257c5363150ccdb1/69_178_3342_2006/master/3342.jpg?width=860&quality=85&auto=format&fit=max&s=2ff73ce46a7d912ab1ac9dd14fc46fad',
    width: 860 },
  { src:
     'https://i.guim.co.uk/img/media/ceaf255ad4460ad68cfe8221257c5363150ccdb1/69_178_3342_2006/master/3342.jpg?width=605&quality=45&auto=format&fit=max&dpr=2&s=a586798d5c1a4070cdfa41503d94a204',
    width: 1210 },
  { src:
     'https://i.guim.co.uk/img/media/ceaf255ad4460ad68cfe8221257c5363150ccdb1/69_178_3342_2006/master/3342.jpg?width=860&quality=45&auto=format&fit=max&dpr=2&s=3bbf7d4c6e780e39c99739c11fcbf73a',
    width: 1720 } ]
``` 

Leading to the following source upgrades.

Screenshot 2020-02-20 at 15 47 13

### CODE REVIEW

File: src/amp/components/topMeta/TopMetaNews.tsx [hunk 1]
```diff
@@ -63,7 +63,9 @@ export const TopMetaNews: React.FC<{
     articleData: ArticleModel;
     adTargeting?: AdTargeting;
 }> = ({ articleData, adTargeting }) => {
-    const {branding} = articleData.commercialProperties[articleData.editionId];
+    const { branding } = articleData.commercialProperties[
+        articleData.editionId
+    ];
```
Comment (jeteve) [inline src/amp/components/topMeta/TopMetaNews.tsx]: This formatting makes me cry, but it makes sense.

Comment (shtukas) [reply src/amp/components/topMeta/TopMetaNews.tsx]: #AutomaticLinter :)

File: src/amp/lib/srcset-utils.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                return acc;
+            }
+            return {
+                items: acc.items.concat(curr),
+                widthsUsed: acc.widthsUsed.concat(curr.width),
+            };
+        },
+        {
+            items: [],
+            widthsUsed: [],
```
Comment (jeteve) [inline src/amp/lib/srcset-utils.ts]: Why not make this an object so testing for membership is more efficient?

Comment (shtukas) [reply src/amp/lib/srcset-utils.ts]: Yeah, I though of using a hash, but in this case, I thought this approach was slightly more natural and any performance penalty is negligible here :)

Comment (jeteve) [reply src/amp/lib/srcset-utils.ts]: ( see http://jsben.ch/eF7Ru )

**Pull request merged**